inappmessaging/src/main/java/com/rakuten/tech/mobile/inappmessaging/runtime/workmanager/workers/DisplayMessageWorker.kt

```diff
@@ -47,15 +46,17 @@ internal class DisplayMessageWorker(
      */
     private fun prepareNextMessage() {
         // Retrieving the next ready message, and its display permission been checked.
-        val message = messageReadinessManager.getNextDisplayMessage() ?: return
+        val messages = messageReadinessManager.getNextDisplayMessage()
         val hostActivity = InAppMessaging.instance().getRegisteredActivity()
-        val imageUrl = message.getMessagePayload().resource.imageUrl
         if (hostActivity != null) {
-            if (!imageUrl.isNullOrEmpty()) {
-                fetchImageThenDisplayMessage(message, hostActivity, imageUrl)
-            } else {
-                // If no image, just display the message.
-                displayMessage(message, hostActivity)
+            for (message in messages) {
```

Comment:
Please confirm if we need to handle normal campaigns on top of tooltip campaigns

Reply:
Retained the current behavior of feat/tooltip-poc, where normal campaigns and tooltip campaigns don't overlap. Will create separate ticket if needed